### PR TITLE
Function which is created by createModifiersMapper handles incorrect inputs

### PR DESCRIPTION
**Pull request by AlexanderShushunov**
Move createModifiersMapper to a separate file.
Add tests.

### CODE REVIEW

File: src/bem-naming-factory/modifires-mapper-creator.spec.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const INVALID_STRING_MODIFIERS = [' ', '*--*', 'кирилица', '4-start-with-digit'];
+const MODIFIERS_WITH_INAPPROPRIATE_TYPE = [22, true, null, /.*/, new Date(), NaN, () => {}, Symbol()];
+
+describe('createModifiersMapper(_name_) should create a function', () => {
+    let modifiersMapper;
+    beforeEach(() => {
+        modifiersMapper = createModifiersMapper('foo');
+    });
+
+    it('which ignores falsey items in the array of modifiers', () => {
```
Comment (redneckz) [inline src/bem-naming-factory/modifires-mapper-creator.spec.js]: Falsy

File: src/bem-naming-factory/modifires-mapper-creator.spec.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+Config.ASSERTION_ENABLED = true;
+const INVALID_STRING_MODIFIERS = [' ', '*--*', 'кирилица', '4-start-with-digit'];
+const MODIFIERS_WITH_INAPPROPRIATE_TYPE = [22, true, null, /.*/, new Date(), NaN, () => {}, Symbol()];
+
+describe('createModifiersMapper(_name_) should create a function', () => {
+    let modifiersMapper;
+    beforeEach(() => {
+        modifiersMapper = createModifiersMapper('foo');
+    });
+
```
Comment (redneckz) [inline src/bem-naming-factory/modifires-mapper-creator.spec.js]: Normal cases should go first. I can see only degenerate cases.

File: src/bem-naming-factory/modifires-mapper-creator.spec.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import {createModifiersMapper} from './modifiers-mapper-creator';
+import {Config} from '../config';
+
+Config.ASSERTION_ENABLED = true;
+const INVALID_STRING_MODIFIERS = [' ', '*--*', 'кирилица', '4-start-with-digit'];
+const MODIFIERS_WITH_INAPPROPRIATE_TYPE = [22, true, null, /.*/, new Date(), NaN, () => {}, Symbol()];
+
+describe('createModifiersMapper(_name_) should create a function', () => {
```
Comment (redneckz) [inline src/bem-naming-factory/modifires-mapper-creator.spec.js]: May be signature check could be represented as separate case.

File: src/bem-naming-factory/modifiers-mapper-creator.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import isEmpty from 'lodash/isEmpty';
+import mapKeys from 'lodash/mapKeys';
+import kebabCase from 'lodash/kebabCase';
+import compact from 'lodash/compact';
+import {assertNamePart} from '../bem-naming-validators';
+import {Config} from '../config';
+
+export function createModifiersMapper(name) {
+    return function mapModifier(modifier) {
+        if (isUndefined(modifier) || (isString(modifier) && isEmpty(modifier)))  {
```
Comment (redneckz) [inline src/bem-naming-factory/modifiers-mapper-creator.js]: After the compact has been added is it really necessary to check undefined.

Comment (redneckz) [inline src/bem-naming-factory/modifiers-mapper-creator.js]: And why the check is limited only by undefined and empty string? What about other empty values?

File: src/bem-naming-factory/modifires-mapper-creator.spec.js (hunk 2 above)
Comment (AlexanderShushunov) [inline src/bem-naming-factory/modifires-mapper-creator.spec.js]: Done

**Pull request merged**